src/main/java/org/mockito/MockitoFramework.java

```diff
@@ -137,4 +137,16 @@ public interface MockitoFramework {
      * @see #clearInlineMocks()
      */
     void clearInlineMock(Object mock);
+
+    /**
+     * Cleans up internal state for all existing mocks. Attempts to interact with mocks after this
+     * is called will throw {@link org.mockito.internal.framework.DisabledMockException} with the
+     * given message.
+     * <p>
+     * This is an alternate solution for the problem that motivated {@link #clearInlineMocks()}, but
+     * also handles the issue of
+     * <a href="https://github.com/mockito/mockito/issues/3419">Issue 3419</a>
+     */
+    void disableInlineMocks(String errorMessage);
```

Comment:
I don't think we should introduce a new API for the behavior here. The feature for `clearInlineMocks` isn't widely used, but we also don't promise the behavior of mocks after clearing of inline mocks. Therefore, I propose we integrate your improvement (good suggestion, I like it!) into `clearInlineMocks`.

In other words: we don't introduce a new API, but instead change the implementation of the existing `clearInlineMocks` and add corresponding tests. This does not pollute our API surface, while improving the usability of this specific API.